core/server/data/migrations/versions/2.4/1-add-webhook-permission-roles.js

```diff
@@ -0,0 +1,55 @@
+const _ = require('lodash');
+const utils = require('../../../schema/fixtures/utils');
+const permissions = require('../../../../services/permissions');
+const logging = require('../../../../lib/common/logging');
+
+const resources = ['webhook'];
+const _private = {};
+
+_private.getPermissions = function getPermissions(resource) {
+    return utils.findModelFixtures('Permission', {object_type: resource});
+};
+
+_private.getRelations = function getRelations(resource) {
+    return utils.findPermissionRelationsForObject(resource);
+};
+
+_private.printResult = function printResult(result, message) {
+    if (result.done === result.expected) {
+        logging.info(message);
+    } else {
+        logging.warn(`(${result.done}/${result.expected}) ${message}`);
+    }
+};
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = (options) => {
+    const localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+
+    return Promise.map(resources, (resource) => {
+        const relationToAdd = _private.getRelations(resource);
+
+        return utils.addFixturesForRelation(relationToAdd, localOptions)
+            .then(result => _private.printResult(result, `Adding permissions_roles fixtures for ${resource}s`))
+            .then(() => permissions.init(localOptions));
+    });
+};
+
+module.exports.down = (options) => {
```

Comment:
Not 100% if we should we have the `down` method in this migration, as previous migrations with permissions/roles didn't add it. Also `down` method here removes both `permissions` and `permissions_roles` for webhooks, is that ideal? 🤷‍♂️

Reply:
Yeah the `down` hook for permissions is tricky, because the fixture utility does add or remove all fixtures for a target resource.

This is tracked here: https://github.com/TryGhost/Ghost/issues/9307

I'd suggest to remove the down function for now.